### PR TITLE
Location distance

### PR DESCRIPTION
**Pull request by Phenomenalhub**

### CODE REVIEW

File: app/src/main/java/com/example/studentarena/MainActivity.java [hunk 1]
```diff
@@ -17,38 +25,72 @@
...
                 return true;
             }
         });
+        setUserLocation();
         bottomNavigationView.setSelectedItemId(R.id.action_home);
     }
+    private void setUserLocation() {
+        LocationManager mLocationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        long UPDATE_INTERVAL = 10000;  /* 10 secs */
+        long MIN_DISTANCE = 1 * 1609; /* 1 mile */
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/MainActivity.java:73]: google tells me 1 mile is 1609.34 meters but i had no idea what that 1609 was. maybe clarify that in the comment?

File: app/src/main/java/com/example/studentarena/MainActivity.java [hunk 2]
```diff
@@ -17,38 +25,72 @@
...
+        int REQUEST_CODE = 1;
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED &&
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_CODE);
+        }
+
+        mLocationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, UPDATE_INTERVAL, MIN_DISTANCE, new LocationListener() {
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/MainActivity.java:81]: so you're updating location every 10 seconds, but only if the update moves a mile or more? it seems like that cause the location manager to never give you updates, unless i'm misunderstanding

to test this, you could post an item with an address at the office and then go for a nice walk (or go home) and keep the app open to see if the displayed distance changes. or did you already do something like that?

**Review by mhammerly — CHANGES_REQUESTED** [review]
functionality looks solid, awesome work! i have some style suggestions and want to know about how you're testing to make sure the location updates and distance measurements are working right, but this is a super meaty feature and it's pretty much done!

**Comment by mhammerly** [comment]
there are still some unaddressed comments here, we can go over them in our sync if you want

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java [hunk 3]
```diff
@@ -49,9 +60,31 @@ public class ComposeFragment extends Fragment {
...
 
-    public ComposeFragment() {
+    public ComposeFragment(MainActivity mainActivity) {
+        activity = mainActivity;
     }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (posted) {
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:75]: i think `posted` has to be set to false again inside this if block

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java [hunk 4]
```diff
@@ -111,13 +177,15 @@ public void done(ParseException e) {
...
+                    posted = true;
+                    activity.getSupportFragmentManager()
                             .beginTransaction()
-                            .replace(R.id.flContainer, new FeedFragment())
+                            .replace(R.id.flContainer, new FeedFragment(activity))
                             .commit();
                 }
             }
         });
+        activity.bottomNavigationView.setSelectedItemId(R.id.action_home);
```
[the remark this answers is not in the document]
Comment (mhammerly) [reply app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java]: so, after posting, `ComposeFragment` would use the fragment manager to go to `FeedFragment`, and then it doesn't ahve to do anything else. i think android will call `FeedFragment`'s `onResume()` method, and you can set the nav bar to home in that method. that way, when goign to a different fragment, you don't' have to remember to also update the nav bar - the act of going to that other fragment will update the nav bar automatically

[the remark this answers is not in the document]
Comment (mhammerly) [reply app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java]: possibly a GitHub Issue to come back to for code cleanup later. can make all your fragments work this way

File: app/src/main/java/com/example/studentarena/BaseActivity.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.example.studentarena;
+
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+//public class BaseActivity extends AppCompatActivity {
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/BaseActivity.java:7]: do you need this file or was this added accidentally?

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java (hunk 3 above)
Comment (mhammerly) [reply app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:77]: reminder about this comment

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java [hunk 6]
```diff
@@ -162,10 +168,11 @@ private void savePost() {
         Post post = new Post();
         post.setDescription(etDescription.getEditText().getText().toString());
         post.setContact(etContactinfo.getEditText().getText().toString());
-        post.setPrice(etPrice.getEditText().getText().toString());
+        // post.setPrice(etPrice.getEditText().getText().toString());
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:171]: should this be commented out? don't you need it to set the price?

File: app/src/main/java/com/example/studentarena/fragments/FeedFragment.java [hunk 7]
```diff
@@ -72,7 +72,7 @@ public void onClick(View v) {
         adapter = new PostsAdapter(getContext(), allPosts, activity);
         // set the adapter on the recycler view
         rvPosts.setAdapter(adapter);
-        queryPosts();
+        //queryPosts();
```
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/FeedFragment.java:75]: is this necessary or not?

File: app/src/main/java/com/example/studentarena/BaseActivity.java (hunk 5 above)
Comment (Phenomenalhub) [reply app/src/main/java/com/example/studentarena/BaseActivity.java:7]: Yes it was added accidentally, I wanted to debug for users permission

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java (hunk 6 above)
Comment (Phenomenalhub) [reply app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:171]: In my search branch I made an edit which causes a conflict here, I changed my price type from string to double. That's why I commented it.

File: app/src/main/java/com/example/studentarena/fragments/FeedFragment.java (hunk 7 above)
Comment (Phenomenalhub) [reply app/src/main/java/com/example/studentarena/fragments/FeedFragment.java:75]: No is not necessary.

**Pull request merged**